config/diff.go

```diff
@@ -112,11 +112,7 @@ func prune(in interface{}) interface{} {
 
 		for k, v := range inType {
 			inType[k] = prune(v)
-			if inType[k] == nil {
-				delete(inType, k)
-			}
```

Comment:
Whats the reasoning behind removing this? This leads to a lot more fields showing up as "null" which is not what we want.

Reply:
This prunes null values, but there are cases when we want to keep them (e.g. originally reported issue when user wants to set ephemeral storage limit to nil).
I agree that we should prune some of the keys - e.g. `external` key if it the value is an empty map.

Reverted to keep deleting keys when the value is e.g. empty map, but to keep them if the value before pruning is nil. Please take another look